Pod/Classes/WPMediaPickerViewController.m

```diff
@@ -36,6 +36,10 @@ @interface WPMediaPickerViewController ()
 @property (nonatomic, assign) BOOL refreshGroupFirstTime;
 @property (nonatomic, strong) UILongPressGestureRecognizer *longPressGestureRecognizer;
 @property (nonatomic, strong) NSIndexPath *assetIndexInPreview;
+
+@property (nonatomic, strong, nullable) Class overlayViewClass;
+@property (nonatomic, strong) NSMutableArray<UIView *> * reusableOverlayViews;
```

Comment:
@frosty do we really need to have our own reusableOverlayViews storage? couldn't create the overlayView we create the dequeue the WPMediaCollectionViewCell? 

So something like this:

```
WPMediaCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:NSStringFromClass([WPMediaCollectionViewCell class]) forIndexPath:indexPath];

if (cell.overlayView == nil || ![cell.overlayView isKindOfClass: overlayViewClass]){
  cell.overlayView = [self.overlayViewClass new];
}
```

This way we will just reuse the cache for cells on the CollectionView, no need to implement ours.

What do you think?

Reply:
So I'm doing our own storage because not every cell might have an overlay view. If the delegate returns `NO` to `shouldDisplayOverlayView`, we won't create one at all. This means if we only have one item uploading, for example, we'd only create one overlay view, instead of having ~28 in cells that we're not using. Does that make sense?